src/Seabug/Metadata/Types.purs

```diff
@@ -0,0 +1,299 @@
+              -- can set a dummy value
+              policyId: wrap
+                $ unsafePartial
+                $ fromJust
+                $ scriptHashFromBytes
+                $ hexToRawBytesUnsafe
+                    "00000000000000000000000000000000000000000000000000000000"
+            , mintPolicy: mempty
+            , collectionNftCS
+            , collectionNftTN
+            , lockingScript
+            , authorPkh
+            , authorShare
+            , marketplaceScript
+            , marketplaceShare
+            , ownerPkh
+            , ownerPrice
+            }
+    where
+    decodePkh :: String -> Either JsonDecodeError PubKeyHash
+    decodePkh =
+      map wrap <<<
+        ( note (TypeMismatch "Invalid Ed25519KeyHash") <<<
+            ed25519KeyHashFromBytes
+            <=< note (TypeMismatch "Invalid ByteArray") <<<
+              hexToRawBytes
+        ) <=< note (TypeMismatch "Expected 0x prefix in authorPkh")
+        <<< removeMetadataBytesPrefix
+
+    decodeShare :: Int -> Either JsonDecodeError Share
+    decodeShare = note (TypeMismatch "Expected int between 0 and 10000")
+      <<< mkShare
+
+    decodeScriptHash :: String -> Either JsonDecodeError ScriptHash
+    decodeScriptHash =
+      note
+        (TypeMismatch "Expected hex-encoded script hash")
+        <<<
+          ( scriptHashFromBytes <<< wrap <=< hexToByteArray <=<
+              removeMetadataBytesPrefix
+          )
+
+    removeMetadataBytesPrefix :: String -> Maybe String
+    removeMetadataBytesPrefix = stripPrefix (Pattern "0x")
+
+newtype SeabugMetadataDelta = SeabugMetadataDelta
+  { policyId :: MintingPolicyHash
+  , ownerPkh :: PubKeyHash
+  , ownerPrice :: Natural
+  }
+
+derive instance Generic SeabugMetadataDelta _
+derive instance Newtype SeabugMetadataDelta _
+derive instance Eq SeabugMetadataDelta
+
+instance Show SeabugMetadataDelta where
+  show = genericShow
+
+instance MetadataType SeabugMetadataDelta where
+  metadataLabel _ = metadataLabel (Proxy :: Proxy SeabugMetadata)
+
+instance ToMetadata SeabugMetadataDelta where
+  toMetadata (SeabugMetadataDelta meta) = toMetadata
+    [ meta.policyId /\
+        [ "ownerPkh" /\ toMetadata meta.ownerPkh
+        , "ownerPrice" /\ toMetadata meta.ownerPrice
+        ]
+    ]
+
+instance FromMetadata SeabugMetadataDelta where
+  fromMetadata (MetadataMap mp) = do
+    policyId /\ contents <- case Map.toUnfoldable mp of
+      [ policyId /\ contents ] ->
+        Tuple <$> fromMetadata policyId <*> pure contents
+      _ -> Nothing
+    ownerPkh <- lookupMetadata "ownerPkh" contents >>= fromMetadata
+    ownerPrice <- lookupMetadata "ownerPrice" contents >>= fromMetadata
+    pure $ SeabugMetadataDelta { policyId, ownerPkh, ownerPrice }
+  fromMetadata _ = Nothing
+
+instance ToData SeabugMetadataDelta where
+  toData (SeabugMetadataDelta meta) = unsafePartial $ toData $ AssocMap.Map
+    [ unsafeMkKey "727" /\ AssocMap.Map
```

Comment:
I believe the key for delta was supposed to be 727d? Unsure if this is permitted in the metadata standard, but it certainly shouldn't be the same as the full metadata, right?

Reply:
Afaict the closest we have to a standard is https://developers.cardano.org/docs/transaction-metadata/, and that won't allow it. Idk what we should change it to though, 728? 727D from base 16 to base 10? I was also [hearing](https://discord.com/channels/917339838808739880/922471807091613747/997252874629300254) something about a new whitepaper for seabug, maybe we can follow that?

Reply:
That was over a week ago, lets ask directly

Reply:
Got no reply, we can leave as is for now since we're not using the delta.

Reply:
I guess it's permitted but using `727d` would save onchain space. Not sure if we're capable of parsing delta metadata now though so lets stick with it